Dcm dev branch gui cont

adambujak opened the pull request.

EAHillebrand committed:
Create README.md

EAHillebrand committed:
Update and rename README.md to thisBranch.md

Continue with Gui

EAHillebrand committed:
updated Documentation

EAHillebrand committed:
Worked on Create User Screen, disabled Admin requirements

adambujak commented on Development/DCM/Common/callbacks.py:
```diff
@@ -1,7 +1,9 @@
 class ApplicationCallbacks:
-    def __init__(self, loginButtonCB, logoffButtonCB, newUserButtonCB, programButtonCB, changeProgramModeCB):
+    def __init__(self, loginButtonCB, logoffButtonCB, newUserButtonCB, programButtonCB, changeProgramModeCB,createUserButtonCB,cancelButtonCB):
```
please fix spacing: item, item, item

adambujak commented on Development/DCM/DCMDatabase/dbpm.py:
```diff
@@ -5,8 +5,7 @@
 from peewee                      import *
-
-from DCMUserAccountManager.duam  import * 
+import DCMUserAccountManager.duam  
```
why did you change this from the existing format?

EAHillebrand replied:
Still attempting to get the class construction on line 62 to work, which use the class User from DCMUserAccountManager.duam

adambujak replied:
those two lines are equivalent though?

adambujak commented on Development/DCM/DCMDatabase/dbpm.py:
```diff
@@ -91,9 +90,10 @@ class BaseModel(Model):
             database = database
 
-    class User(BaseModel):
+    class User1(BaseModel):
```
rename to something meaningful please

adambujak commented on Development/DCM/DCMGraphicalUserInterface/guic.py:
```diff
@@ -13,6 +13,7 @@
 class ScreenNames(Enum):
     LOGIN_SCREEN        = 0
     PROGRAMMING_SCREEN  = 1
+    CREATEUSER_SCREEN   = 2
```
RENAME to CREATE_USER_SCREEN

adambujak commented on Development/DCM/DCMGraphicalUserInterface/guic.py:
```diff
@@ -43,7 +52,7 @@ def __init__(self, screenName, data):
         C_LOGIN_PASSWORD_LABEL, 
         C_LOGIN_BUTTON_TEXT, 
-        C_LOGIN_NEW_USER_BUTTON_TEXT))
+        C_NEW_USER_BUTTON_TEXT))
```
nice

adambujak commented on Development/DCM/DCMGraphicalUserInterface/guic.py:
```diff
@@ -117,7 +140,7 @@ def p_drawLoginScreen(self, data):
     def p_drawProgrammingScreen(self, data):
         """
-        Draw screen of tpye programming screen
+        Draw screen of type programming screen
```
nice catch

adambujak commented on Development/DCM/DCMGraphicalUserInterface/guic.py:
```diff
@@ -99,6 +118,10 @@ def getLoginData(self):
             entryData = self.gui.getEntryData()
             return LoginData(entryData[0], entryData[1])
 
+    def getNewUserData(self):
```
add function brief by using triple quotes

adambujak commented on Development/DCM/DCMGraphicalUserInterface/guic.py:
```diff
@@ -126,8 +149,15 @@ def p_drawProgrammingScreen(self, data):
             data.buttonTexts,
             data.buttonCallbacks)
+
+    def p_drawCreateUserScreen(self, data):  #create User and cancel
```
function brief

adambujak commented on Development/DCM/DCMGraphicalUserInterface/readme.txt:
```diff
@@ -1,2 +1,5 @@
-guic.py is the Graphical User Interface Controller source file
\ No newline at end of file
+guic.py is the Graphical User Interface Controller source file
+guial.py is the Graphical User Interface Abstraction Layer Class
```
thanks for updating

adambujak commented on Development/DCM/DCMUserAccountManager/duam.py:
```diff
@@ -87,12 +87,14 @@ def signInUser(self, p_loginData):
             usernameValid = (userData.getUsername() == username)
-
+            print(userData.getUsername() + userData.getPassword())
+            print(passwordValid)
```
what's the point of these? if debugging, it's a good idea to put a label on it - print("passwordValid:", passwordValid)

adambujak commented on Development/DCM/main.py:
```diff
@@ -25,19 +25,27 @@ def updateGUI(self):
+        self.guiController.drawScreen(createUserScreen)
 
+    def createUserButtonCB(self):
+        x = self.accountController.makeNewUser(self.guiController.getNewUserData(),"hello")
```
is hello the admin password? if so, please change it to "adminPassword" so that it's not just a random hello in the code. Probably best to just add it as a constant

adambujak commented on Development/DCM/main.py:
```diff
@@ -48,9 +56,16 @@ def logoffButtonCallback():
+
 
-    callbacks = ApplicationCallbacks(loginButtonCallback, logoffButtonCallback, newUserButtonCallback, None, None)
+    callbacks = ApplicationCallbacks(loginButtonCallback, logoffButtonCallback, newUserButtonCallback, None, None,createUserButtonCallback,cancelButtonCallback)
```
spacing

adambujak commented on Development/DCM/src/dcm_constants.py:
```diff
@@ -30,8 +30,8 @@
-C_LOGIN_BUTTON_TEXT            =       "Sign In"                                          # BUTTON TEXT VALUE FOR LOGIN BUTTON
-C_LOGIN_NEW_USER_BUTTON_TEXT   =       "New User"                                         # BUTTON TEXT VALUE FOR NEW USER BUTTON
+C_LOGIN_BUTTON_TEXT            =       "Login"                                          # BUTTON TEXT VALUE FOR LOGIN BUTTON
+C_NEW_USER_BUTTON_TEXT   	   =       "New User?"                                         # BUTTON TEXT VALUE FOR NEW USER BUTTON
```
fix alignment of comments please

adambujak commented on Development/DCM/src/dcm_constants.py:
```diff
@@ -40,4 +40,7 @@
\ No newline at end of file
+C_PROGRAM_LOGOUT_BUTTON_TEXT   =       "Sign Out"                                         # BUTTON TEXT VALUE FOR LOGOUT BUTTON
+
+C_CREATE_USER_BUTTON_TEXT	   = 	   "Create New User"
```
comments

EAHillebrand replied:
I dont see this issue in my text editor

EAHillebrand committed:
Worked on signOut. User class issue resolved by deffining class in dbpm.py. Cannot Login

adambujak commented on Development/DCM/DCMGraphicalUserInterface/guic.py:
```diff
@@ -119,7 +119,10 @@ def getLoginData(self):
 
     def getNewUserData(self):
-        if self.currentScreen.screenName == ScreenNames.CREATEUSER_SCREEN:
+        '''
```
Either use double quotes everywhere, or single quotes everywhere please

adambujak commented on Development/DCM/DCMDatabase/dbpm.py:
```diff
@@ -5,15 +5,26 @@
+class User:
+    def __init__(self, p_username, p_password, p_userRole):
+        self.username    = p_username
+        self.password    = p_password
```
this shouldn't be here.. this doesn't relate to the database user data, this relates to the user in the user account manager

adambujak commented on Development/DCM/main.py:
```diff
@@ -25,27 +25,35 @@ def updateGUI(self):
 
 def main():
-    #print(hash_password('hello')) #Displays hashed Admin password
+    #print(hash_password("C_ADMIN_PASSWORD")) #Displays hashed C_ADMIN_PASSWORD
```
C_ADMIN_PASSWORD may be confusing since it follows the same scheme as our constant variable names

adambujak committed:
Fixed user log in bug

adambujak committed:
Fixed Formatting (alignment)

EAHillebrand committed:
Added remaining user fields on program screen. Implementing gets for parameters, still need to work on database side of it.

EAHillebrand committed:
Implemented User Cap

adambujak committed:
Added user programming data storage feature

stephen99scott committed:
Added documentation for hardware

Includes the documentation for part 1, 3, bonus and testing

stephen99scott committed:
Added model which includes AOO, VOO, AAI and VVI

This model includes all required modes for assignment 1 with annotations and implementation of the bonus

EAHillebrand committed:
Updated GUI for program button callback, Rearranged Program Screen

stephen99scott committed:
Merge pull request #2 from adambujak/SimulinkDevBranch

Simulink Model Development Branch

stephen99scott committed:
Added the most recent version of all documentation

stephen99scott committed:
Added the zip folder for assignment 1

Contains the most up-to-date files for the submission of assignment 1

EAHillebrand committed:
Implemented validation for program screen user input

adambujak committed:
Added form filling feature
Changed action on login - now it goes back to log in menu
Added default values for programming data

EAHillebrand committed:
Displayed Error messages to UI. Moved class FailureCodes to common from duam

EAHillebrand committed:
Added testing_group5_Final doc to Documentation folder

EAHillebrand committed:
fixed bug in main.py

stephen99scott committed:
Added the final assignment 1 zip

This is the final submission that was submitted by Eric on Avenue

stephen99scott committed:
Added updated DCM documentation

Includes the changes for GUI testing

EAHillebrand committed:
Implemented saving program mode to database

EAHillebrand committed:
Updated some documentation for testing

EAHillebrand committed:
congifured Programming screen to only display relevant parameters

EAHillebrand committed:
Merge branch 'master' into DCMDevBranchGuiCont
